### PR TITLE
Remove odo preference registry update command

### LINKED ISSUE #5535: odo preference - cleanup
there should be the following set of subcommands for `odo preference` command

- [x] `odo preference set` - sets value for single value item (`UpdateNotification`, `Timeout`, `PushTimeout`, `RegistryCacheTime`, `Ephemeral`, `ConsentTelemetry`) (#5822 )
- [x] `odo preference unset` - unsets value (restores it to default) for single value item (`UpdateNotification`, `Timeout`, `PushTimeout`, `RegistryCacheTime`, `Ephemeral`, `ConsentTelemetry`) (#5822 )
- [x] `odo preference add` - adds value to array items (`registry`) (#5873 )
- [x] `odo preference remove` - removes value from array items (`registry`) (#5873 )
- [x] `odo preference view` -  view all preferences, and move `registry list` to `view` (#5850)
- [x] review name of the preference items, if needed rename it  and add description (#5822 )
- [x] Drop odo preference registry update; if needed, user can always remove and add registry (#5853)
## examples 
```
$ odo preference set ConsentTelemetry true

Value of 'ConsentTelemetry' preference was set to 'true'
```

```
$ odo preference unset ConsentTelemetry

Value of 'ConsentTelemetry' was removed from preferences. Its default value will be used.
```

```
$ odo preference add registry StagingRegistry https://registry.stage.devfile.io```

New Devfile registry was added to odo.
```

```
$ odo preference remove registry StagingRegistry

Devfile registry 'StagingRegistry' was removed from odo.
```

```
$ odo preference view

Preference parameters:
PARAMETER             VALUE
UpdateNotification    true (default)
Timeout               
PushTimeout           
RegistryCacheTime     
Ephemeral             
ConsentTelemetry      false


Devfile registries:
NAME                       URL                                   
StagingRegistry            https://registry.stage.devfile.io     
DefaultDevfileRegistry     https://registry.devfile.io           
``

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**

**What does this PR do / why we need it:**
This PR removes `odo preference registry update` command.
**Which issue(s) this PR fixes:**

Fixes part of #5535 

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Review by cdrage — APPROVED** [review]
/lgtm

**Comment by dharmit** [comment]
> Fixes #5535

Does this PR completely fix the said issue or a part of it?

**Comment by valaparthvi** [comment]
> > Fixes #5535
> 
> Does this PR completely fix the said issue or a part of it?

Fixed it. thanks for catching it!

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (o RegistryClient) ListDevfileStacks(registryName, devfileFlag, filterFlag
 		retrieveRegistryIndices.Add(util.ConcurrentTask{ToRun: func(errChannel chan error) {
 			registryDevfiles, err := getRegistryStacks(o.preferenceClient, registry)
 			if err != nil {
-				log.Warningf("Registry %s is not set up properly with error: %v, please check the registry URL and credential (refer `odo preference registry update --help`)\n", registry.Name, err)
+				log.Warningf("Registry %s is not set up properly with error: %v, please check the registry URL, and credential and add the registry again (refer `odo preference registry --help`)\n", registry.Name, err)
```
Comment (rm3l) [inline pkg/registry/registry.go:123]: `refer` should be intransitive here, no?

```suggestion
				log.Warningf("Registry %s is not set up properly with error: %v, please check the registry URL, and credential and add the registry again (refer to `odo preference registry --help`)\n", registry.Name, err)
```

File: docs/website/versioned_docs/version-3.0.0/overview/configure.md [hunk 2]
```diff
@@ -144,21 +144,4 @@ Successfully deleted registry
 
 You can use the `--force` (or `-f`) flag to force the deletion of the registry without confirmation.
 
-### Updating a registry
```
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/overview/configure.md:147]: Wondering if we should not keep this section, and just mention that users need to delete the registry and add it again... What do you think?

**Comment by feloy** [comment]
/override ci/prow/v4.10-integration-e2e 

Tests pass on IBM Cloud

**Pull request merged**